spec/models/assignment_spec.rb

```diff
@@ -2227,6 +2227,42 @@ def grouping_count(groupings)
     end
   end
 
+  describe '#get_num_valid' do
+    before :each do
+      @assignment = create(:assignment, assignment_properties_attributes: { group_min: 2, group_max: 3 })
+      @groupings = Array.new(4) { create(:grouping, assignment: @assignment) }
+    end
+
+    context 'When two groups meet min size requirement' do
+      before :each do
+        create(:accepted_student_membership, grouping: @groupings.first)
+        create(:inviter_student_membership, grouping: @groupings.first)
+        create(:student_membership, grouping: @groupings.second)
+        create(:student_membership, grouping: @groupings.second)
+      end
+      it 'should return # of valid groups that meet size requirement' do
+        expect(@assignment.get_num_valid).to eq(2)
+      end
+    end
+
+    context 'When one group meets min size requirement and another is instructor approved' do
+      before :each do
+        create(:accepted_student_membership, grouping: @groupings.first)
+        create(:inviter_student_membership, grouping: @groupings.first)
+        @groupings.second.update_attribute(:instructor_approved, true)
```

Comment:
Use [`update`](https://guides.rubyonrails.org/active_record_basics.html#update) instead of `update_attribute`.